### PR TITLE
Add support for BL652 & DVK

### PR DESCRIPTION
**Pull request by BWhitten**
Adding basic support for the BL652 & DVK from Laird
I have also included the SD download folders to the .gitignore list.

### CODE REVIEW

**Comment by glennrub** [comment]
Thanks a lot for the contribution! 

The patch / board support looks very good. 
I have one only one comment and i will address that in the code. 

We are in the middle of doing a PR towards upstream micropython master. And i guess we should try to get this in first.

File: nrf/boards/dvk_bl652/mpconfigboard.h [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+#define MICROPY_HW_SPI0_NAME        "SPI0"
+#define MICROPY_HW_SPI0_SCK         (pin_A25)
+#define MICROPY_HW_SPI0_MOSI        (pin_A23)
+#define MICROPY_HW_SPI0_MISO        (pin_A24)
+
+#define MICROPY_HW_PWM0_NAME        "PWM0"
+#define MICROPY_HW_PWM1_NAME        "PWM1"
+#define MICROPY_HW_PWM2_NAME        "PWM2"
+
+#define HELP_TEXT_BOARD_LED         "1,2,3,4"
```
Comment (glennrub) [inline nrf/boards/dvk_bl652/mpconfigboard.h]: This is a macro used in help.c, which indicates the LED number options available. Changing this to "1, 2" and it should be good.

Comment (BWhitten) [reply nrf/boards/dvk_bl652/mpconfigboard.h]: Thanks, good catch.

File: .gitignore [hunk 2]
```diff
@@ -42,3 +42,4 @@ user.props
 # Generated rst files
 ######################
 genrst/
+
```
Comment (glennrub) [inline .gitignore]: Does this patch have any purpose? Should we let it go to have less diff against upstream master as it is outside the nrf/ folder?

Comment (BWhitten) [reply .gitignore]: Can be dropped, my setup generally complains about no newlines at the end of files.
I shall update.

**Comment by glennrub** [comment]
Great! Looks perfect now :) Thanks!

**Pull request merged**